stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -175,7 +175,13 @@ end
 (*)(D::Diagonal, x::Number) = Diagonal(D.diag * x)
 (/)(D::Diagonal, x::Number) = Diagonal(D.diag / x)
 (*)(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag .* Db.diag)
```

Comment:
This method needs the same treatment -- can you add that too?

Reply:
Sure, I pushed another commit

Reply:
I don't think that Diagonal*Diagonal needs it. It falls back to the element wise product of vectors which has a dimensionality check.

Reply:
It seems that a single-element vector behaves as a scalar when doing the element-wise product:
```
julia> randn(1) .* randn(5)
5-element Vector{Float64}:
 -0.09149628486880286
 -0.20889567435214648
  0.09612350337701256
  0.08092068173428073
  0.06652421890696196
```

Reply:
Ah, nice catch.